user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/BatchRegistrationDialog2.java

```diff
@@ -0,0 +1,547 @@
+    for (ExperimentalGroup experimentalGroup : experimentalGroups) {
+      List<BiologicalReplicate> sortedReplicates = experimentalGroup.biologicalReplicates().stream()
+          .sorted(Comparator.comparing(BiologicalReplicate::label))
+          .toList();
+      for (BiologicalReplicate biologicalReplicate : sortedReplicates) {
+        // new sampleInfo
+        SampleInfo sampleInfo = new SampleInfo();
+        sampleInfo.biologicalReplicate = biologicalReplicate;
+        sampleInfo.experimentalGroup = experimentalGroup;
+        if (species.size() == 1) {
+          sampleInfo.setSpecies(species.get(0));
+        }
+        if (specimen.size() == 1) {
+          sampleInfo.setSpecimen(specimen.get(0));
+        }
+        if (analytes.size() == 1) {
+          sampleInfo.setAnalyte(analytes.get(0));
+        }
+        sampleInfos.add(sampleInfo);
+      }
+    }
+    return sampleInfos;
+  }
+
+  private static String formatConditionString(Condition condition) {
+    return condition.getVariableLevels().parallelStream()
+        .map(variableLevel -> "%s:%s %s"
+            .formatted(variableLevel.variableName().value(),
+                variableLevel.experimentalValue().value(),
+                variableLevel.experimentalValue().unit().orElse(""))
+            .trim())
+        .sorted()
+        .collect(Collectors.joining("; "));
+  }
+
+  private static ComponentRenderer<Span, AnalysisMethod> getAnalysisMethodItemRenderer() {
+    return new ComponentRenderer<>(analysisMethod -> {
+      var listItem = new Span();
+      listItem.addClassName("spreadsheet-list-item");
+      Span label = new Span(analysisMethod.label());
+      label.setText(analysisMethod.label());
+      var questionMarkIcon = VaadinIcon.QUESTION_CIRCLE_O.create();
+      questionMarkIcon.setTooltipText(analysisMethod.description());
+      listItem.add(label, questionMarkIcon);
+      return listItem;
+    });
+  }
+
+  private void onBatchNameChanged(
+      ComponentValueChangeEvent<TextField, String> batchNameChangedEvent) {
+    /* do nothing */
+  }
+
+  private void onRemoveLastRowClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.removeLastRow();
+  }
+
+  private void onAddRowClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.setValidationMode(ValidationMode.LAZY);
+    spreadsheet.addRow(new SampleInfo());
+    spreadsheet.setValidationMode(ValidationMode.EAGER);
+  }
+
+  private void onPrefillClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.setValidationMode(ValidationMode.LAZY);
+    spreadsheet.resetRows();
+    for (SampleInfo sampleInfo : prefilledSampleInfos()) {
+      spreadsheet.addRow(sampleInfo);
+    }
+    spreadsheet.setValidationMode(ValidationMode.EAGER);
+  }
+
+  @Override
+  protected void onConfirmClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.validate();
+    if (spreadsheet.isInvalid()) {
+      return;
+    }
+    if (batchNameField.isInvalid()) {
+      return;
+    }
```

Comment:
Should there be a notification shown to the user informing him that there are invalid fields?

Reply:
As the fields are highlighted and I did not get this input, this is up to the user tests.